_posts/2016-01-14-javascript-code-coverage-with-instanbul-and-coveralls.markdown

```diff
@@ -0,0 +1,119 @@
+---
+layout: post
+title: "JavaScript Code Coverage With Instanbul and Coveralls"
+date: 2016-01-14T21:50:18-02:00
+keywords: test,
+author: rafaeleyng
+excerpt: >
+  A fast lane to set up code coverage in your project
+---
+
+## Code coverage
+
+If you want to set up code coverage measurements in a project, there are usually 4 things that you'll need:
+
+- a **test runner** (like [https://github.com/mochajs/mocha](Moccha), [AVA](https://github.com/sindresorhus/ava)), to run your tests
+- a **code coverage** tool (like [https://github.com/gotwarlost/istanbul](Istanbul), [https://github.com/bcoe/nyc](nyc)), to generate code coverage reports
+- a **code coverage insight** service (like [https://coveralls.io/](Coveralls), [https://codecov.io/](Codecov)), to provide you a nice visualization of your code coverage data
+- a **reporting** tool (like [https://github.com/nickmerwin/node-coveralls](node-coveralls), [https://github.com/cainus/codecov.io/](codecov.io)), to send your reports to some service
+
+## Assumptions
+
+I'll assume you already have Mocha locally installed (`npm i -D mocha`) as your test runner, and show you how to do the rest.
+
+I'll also assume that you have TravisCI server configured for your project.
+
+### npm scripts
+
+npm scripts are scripts you add inside package.json. The "scripts" object contains your scripts. You can add arbitrary shell script commands there, like `"print-something": "echo 'something'"`. You run this script by invoking it by its name: `npm run print-something`.
```

Comment:
The "scripts" object contains your scripts.

> npm scripts are scripts you add inside package.json. The "scripts" object contains your scripts. You can add arbitrary shell script commands there, like `"print-something": "echo 'something'"`. You run this script by invoking it by its name: `npm run print-something`.

Another way to say that: 
 npm scripts are scripts you add inside package.json You can add arbitrary shell script commands there, like:
 `"scripts: { "print-something": "echo 'something'" }`. You run this script by invoking it by its name: `npm run print-something`.